ExceptionHandling/Task2/NumberParser.cs

```diff
@@ -1,12 +1,77 @@
 ﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.SymbolStore;
+using System.Linq;
 
 namespace Task2
 {
     public class NumberParser : INumberParser
     {
+        private const char Plus = '+';
+        private const char Minus = '-';
+
+        private readonly static Dictionary<char, int> ValidNumbers = new Dictionary<char, int>
+        {
+            { '0', 0 },
+            { '1', 1 },
+            { '2', 2 },
+            { '3', 3 },
+            { '4', 4 },
+            { '5', 5 },
+            { '6', 6 },
+            { '7', 7 },
+            { '8', 8 },
+            { '9', 9 }
+        };
+
         public int Parse(string stringValue)
         {
-            throw new NotImplementedException();
+            if (stringValue == null)
+            {
+                throw new ArgumentNullException(nameof(stringValue));
+            }
+
+            if (string.IsNullOrWhiteSpace(stringValue))
+            {
+                throw new FormatException("Input value is empty.");
+            }
+
+            var isMinus = false;
+            long numericValue = 0;
+
+            for (int i = 0; i < stringValue.Trim().Length; i++)
```

Comment:
it would be better to trim before iterating and get values from the trimmed string

Reply:
Fixed